server/routes/createGoal/createGoalController.ts

```diff
@@ -83,8 +83,9 @@ export default class CreateGoalController {
     req.session.newGoal.addStepForms = req.session.newGoal.addStepForms || []
 
     if (isEditMode(req)) {
-      req.session.newGoal.addStepForm =
-        req.session.newGoals[parseInt(goalIndex, 10) - 1].addStepForms[parseInt(stepIndex, 10) - 1]
+      req.session.newGoal.addStepForm = req.session.newGoals[parseInt(goalIndex, 10) - 1].addStepForms[
```

Comment:
I know I helped you with this earlier, but I wonder if we can change it slightly to make it a little easier to read. How about something like:
```
      const newGoalForm =  req.session.newGoals[parseInt(goalIndex, 10) - 1]
      let addStepForm = newGoalForm.addStepForms[parseInt(stepIndex, 10) - 1]
      // In the case of adding a new step in edit mode there won't be an existing AddStepForm on the session so we need to instantiate one
      if (!addStepForm) {
        addStepForm = { stepNumber: parseInt(stepIndex, 10) }
      }
      req.session.newGoal.addStepForm = addStepForm
```

I know its quite long and wordy (and perhaps there's a more concise way of expressing it?), but I think its perhaps more readable like this?

What do you think?

Reply:
Makes sense and more readable

Reply:
Having said that .... I'm not sure now ... (sorry! 😁 )

What I'm not sure about is whether this code needs to exist here at all .... its hard to explain in text so I'll check out your branch and have a tinker, and then perhaps we can do a little pairing so I can better explain my thought .... give me a few mins ....